core/src/modules/ops/recordProperty.ts

```diff
@@ -228,10 +233,12 @@ async function preparePropertyImports(
   );
 
   if (method === "RecordProperties") {
-    await CLS.enqueueTask(`recordProperty:importRecordProperties`, {
-      propertyIds: properties.map((p) => p.id),
-      recordIds: uniqueRecordIds,
-    });
+    while (uniqueRecordIds.length > 0) {
+      await CLS.enqueueTask(`recordProperty:importRecordProperties`, {
+        propertyIds: properties.map((p) => p.id),
+        recordIds: uniqueRecordIds.splice(0, limit),
+      });
+    }
```

Comment:
On staging, the task that was running out of memory looked like this:

```
debug: [ worker ] working job recordProperties workerId=1 class=recordProperty:importRecordProperties queue=recordProperties args={"propertyIds":["rul_fa41c80c-7a31-449a-9be6-fd5e0ebd58eb","rul_29cb27cb-c696-4527-929f-e302b969a423","rul_b7e3c82a-5ae2-4fc9-a0b4-f9e0eb319e29","rul_d5dbb7af-56f0-47c5-a4c8-128eb724b4c5","gender","rul_2a0e0039-9360-4b95-a1d0-75582841e23f","rul_85a8d395-080f-4974-b3db-36e59d2001b9","ip_address","vip"],"recordIds":"2997 items"}
```

That's a reasonable number of properties for an unreasonable number of records.  This change enqueues a few jobs never to exceed the `limit` number of records at a time.

Reply:
This is a rather timely addition to https://github.com/grouparoo/grouparoo/pull/2817 (cc @bleonard)